apps/prairielearn/src/lib/aws.js

```diff
@@ -216,38 +210,36 @@ module.exports.downloadFromS3 = util.callbackify(module.exports.downloadFromS3As
  * @param {boolean=} isDirectory - Whether the deletion target is a directory (defaults to false).
  */
 module.exports.deleteFromS3Async = async function (s3Bucket, s3Path, isDirectory = false) {
-  const s3 = new AWS.S3();
+  const s3 = new S3(module.exports.getS3ClientConfig());
 
   if (isDirectory) {
     s3Path += s3Path.endsWith('/') ? '' : '/';
   }
-
-  const params = {
+  await s3.deleteObject({
     Bucket: s3Bucket,
     Key: s3Path,
-  };
-  await s3.deleteObject(params).promise();
+  });
   debug(`Deleted s3://${s3Bucket}/${s3Path}`);
 };
+module.exports.deleteFromS3 = util.callbackify(module.exports.deleteFromS3Async);
 
 /**
  * Get a file from S3.
  *
  * @param {string} bucket - S3 bucket name.
  * @param {string} key - The S3 target path.
  * @param {boolean} buffer - Defaults to true to return buffer.
- * @return {Promise<AWS.S3.Body>} Buffer or ReadableStream type from S3 file contents.
+ * @return {Promise<Buffer | import('@aws-sdk/client-s3').GetObjectOutput['Body']>} Buffer or ReadableStream type from S3 file contents.
  */
 module.exports.getFromS3Async = async function (bucket, key, buffer = true) {
```

Comment:
I don't particularly like this "third argument dictates return type" thing, we should make this just return `Body` directly in the future.

Reply:
Agreed. This is horrible now that we have typing.